fog/types/src/common.rs

```diff
@@ -61,6 +61,51 @@ impl BlockRange {
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    /// Merge the first contiguous ranges into a range
+    ///
+    /// The ranges should be sorted to produce consistent results.
+    /// They are not sorted here in order to properly indicate the index of the
+    /// last range merged.
```

Comment:
I'm up for suggestions on a better description here.

Reply:
```suggestion
    /// The "ranges" input must be pre-sorted by increasing start_block to produce correct results.
    /// Sorting is left to the caller so as to return a meaningful "index" which indicates an
    /// input range that contains an "end_block" at the top of the first contiguous range of blocks.
```